leash.go

```diff
@@ -605,3 +621,93 @@ func logStats(stats *responseStats, interval uint) {
 		stats.logAndReset()
 	}
 }
+
+func getBaseTime(options GlobalOptions) (time.Time, error) {
+	var baseTime time.Time
+
+	// support multiple files and globs, although this is unlikely to be used
+	searchFiles := []string{}
+	for _, f := range options.Reqs.LogFiles {
+		// can't work with stdin
+		if f == "-" {
+			continue
+		}
+		// can't work with files that don't exist
+		if files, err := filepath.Glob(f); err == nil && files != nil {
+			searchFiles = append(searchFiles, files...)
+		}
+	}
+	if len(searchFiles) == 0 {
+		return baseTime, fmt.Errorf("unable to get base time, no files found")
+	}
+
+	// we're going to have to parse lines, so get an instance of the parser
+	parser, parserOpts := getParserAndOptions(options)
+	parser.Init(parserOpts)
+	lines := make(chan string)
+	events := make(chan event.Event)
+	var prefixRegex *parsers.ExtRegexp
+	if options.PrefixRegex == "" {
+		prefixRegex = nil
+	} else {
+		prefixRegex = &parsers.ExtRegexp{regexp.MustCompile(options.PrefixRegex)}
+	}
+	// read each file, throw the last line on the lines channel
+	go getEndLines(searchFiles, lines)
+	// the parser will parse each line and give us an event
+	go func() {
+		// ProcessLines will stop when the lines channel closes
+		parser.ProcessLines(lines, events, prefixRegex)
+		// Signal that we're done sending events
+		close(events)
+	}()
+
+	// we read the event and find the latest timestamp
+	// this is our base time (assuming the input files are sorted by time,
+	// otherwise we'd have to parse *everything*)
+	for ev := range events {
+		if ev.Timestamp.After(baseTime) {
+			baseTime = ev.Timestamp
+		}
+	}
+
+	return baseTime, nil
+}
+
+func getEndLines(files []string, lines chan<- string) {
+	for _, f := range files {
+		lines <- getEndLine(f)
+	}
+
+	close(lines)
+}
+
+func getEndLine(file string) string {
+	handle, err := os.Open(file)
+	if err != nil {
+		logrus.WithError(err).WithField("file", file).
+			Fatal("unable to open file")
+	}
+	defer handle.Close()
+	// we use a scanner to read to the last line
+	// not the most efficient
```

Comment:
this is going to be super bad on a multi-gigabyte file. Can we open the file at the end then read back a reasonable amount and start scanning there instead?